pyiron_base/jobs/dynamic.py

```diff
@@ -132,4 +132,53 @@ def _get_template_classes():
     }
 
 
+def create_new_job_type(
+    class_name, input_dict, write_input_funct, collect_output_funct, executable_str
+):
+    from pyiron_base.jobs.job.template import TemplateJob
+
+    def _get_write_input_func(write_input_funct):
```

Comment:
Is the double nesting necessary?  In my quick testing, just defining `write_input` immediately in the function body works as well.